website/docs/sdk-reference/rust.mdx

```diff
@@ -0,0 +1,705 @@
+
+```rust
+use std::time::Duration;
+use configcat::{Client, PollingMode, DataGovernance};
+
+let builder = Client::builder("#YOUR-SDK-KEY#")
+    .polling_mode(PollingMode::AutoPoll(Duration::from_secs(60)))
+    .data_governance(DataGovernance::EU);
+
+let client = builder.build().unwrap();
+```
+
+Available options:
+
+| Option                              | Description  |
+| ----------------------------------- | ------------ |
+| `data_governance(DataGovernance)`   | Defaults to `Global`. Describes the location of your feature flag and setting data within the ConfigCat CDN. This parameter needs to be in sync with your Data Governance preferences. [More about Data Governance](../advanced/data-governance.mdx). Available options: `Global`, `EuOnly`. |
+| `base_url(&str)`                    | Sets the CDN base url (forward proxy, dedicated subscription) from where the sdk will download the configurations. |
+| `cache(Box<dyn ConfigCache>)`       | Sets a custom cache implementation for the client. [See below](#cache). |
+| `http_timeout(Duration)`            | Sets the maximum wait time for a HTTP response. [More about the HTTP timeout](#http-timeout) |
+| `polling_mode(PollingMode)`         | Defaults to `AutoPoll`. Sets the polling mode for the client. [More about polling modes](#polling-modes). |
+| `overrides(Box<dyn OverrideDataSource>, OverrideBehavior)` | Sets the local feature flag & setting overrides. [More about feature flag overrides](#flag-overrides). |
+| `default_user(User)`                | Sets the default user. [More about default user](#default-user). |
+| `offline(bool)`                     | Defaults to `false`. Indicates whether the SDK should be initialized in offline mode. [More about offline mode](#online--offline-mode). |
+
+:::caution
+We strongly recommend you to use the _ConfigCat Client_ as a Singleton object in your application.
+If you want to use multiple SDK Keys in the same application, create only one _ConfigCat Client_ per SDK Key.
+:::
+
+## Anatomy of `get_value()`
+
+| Parameters     | Description                                                                                                  |
+| -------------- | ------------------------------------------------------------------------------------------------------------ |
+| `key`          | **REQUIRED.** Setting-specific key. Set on _ConfigCat Dashboard_ for each setting.                           |
+| `user`         | Optional, _User Object_. Essential when using Targeting. [Read more about Targeting.](../targeting/targeting-overview.mdx) |
+| `default`      | **REQUIRED.** This value will be returned in case of an error.                                               |
+
+```rust
+let is_awesome_feature_enabled = client.get_value(
+    "isAwesomeFeatureEnabled", // Setting Key
+    Some(User::new("#UNIQUE-USER-IDENTIFIER#")), // Optional User Object
+    false // Default value
```

Comment:
Just out of curiosity: why are the `defaultValue` and `user` parameters passed to the `get_value` method in reverse order compared to the other SDKs?

If there's no solid reason behind this, I think it would make more sense to make the `user` parameter the last one - not only for consistency, but also because the `user` parameter is not necessarily required to evaluate a feature flag, i.e. it may be omitted. (Ok, as far as I can see, Rust doesn't support default function arguments at the moment, but it's a language feature that's usually added eventually...)

Reply:
Done